### PR TITLE
uid-range: several cleanups

### PR DESCRIPTION
**Pull request by yuwata**

### CODE REVIEW

File: src/basic/uid-range.c [hunk 1]
```diff
@@ -22,10 +22,28 @@ static bool uid_range_intersect(UidRange *range, uid_t start, uid_t nr) {
...
+                return r;
+
+        return CMP(a->nr, b->nr);
+}
+
 static void uid_range_coalesce(UidRange **p, size_t *n) {
         assert(p);
         assert(n);
 
+        if (*n == 0)
```
Comment (poettering) [inline src/basic/uid-range.c]: might as well do: `*b <= 1`

File: src/basic/uid-range.h [hunk 2]
```diff
@@ -8,7 +8,10 @@ typedef struct UidRange {
         uid_t start, nr;
 } UidRange;
 
-int uid_range_add(UidRange **p, size_t *n, uid_t start, uid_t nr);
+int uid_range_add_internal(UidRange **p, size_t *n, uid_t start, uid_t nr, bool coalsece);
```
Comment (poettering) [inline src/basic/uid-range.h]: typo in coalesce

File: src/basic/uid-range.c [hunk 3]
```diff
@@ -214,13 +195,20 @@ int uid_range_load_userns(UidRange **p, size_t *n, const char *path) {
...
                                 return errno_or_else(EIO);
 
-                        return 0;
+                        break;
                 }
                 if (k != 3)
                         return -EBADMSG;
 
-                r = uid_range_add(p, n, uid_base, uid_range);
+                r = uid_range_add(&q, &m, uid_base, uid_range);
```
Comment (poettering) [inline src/basic/uid-range.c]: i figure you wanted to change this to uid_range_add_internal() with coalesce=false, no?

File: src/basic/uid-range.h [hunk 4]
```diff
@@ -8,7 +8,10 @@ typedef struct UidRange {
...
+static inline int uid_range_add(UidRange **p, size_t *n, uid_t start, uid_t nr) {
+        return uid_range_add_internal(p, n, start, nr, true);
+}
```
Comment (poettering) [inline src/basic/uid-range.h:24]: hmm, why not make uid_range_add_internal() a static function, and drop the coalesce parameter? And then make uid_range_add() a regular public function that calls uid_range-add_internal() and then itself calls uid_range_coalesce()?

Comment (yuwata) [reply src/basic/uid-range.h:24]: That makes sense. but in the revised version several tests for `uid_range_coalesce()` are added. See the last commit.

**Comment by yuwata** [comment]
@poettering Thank you for the review. Updated. PTAL.

**Pull request merged**